apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -460,32 +461,45 @@ const aichatSlice = createSlice({
 });
 
 const hasFilledOutModelCard = (modelCardInfo: ModelCardInfo) => {
-  for (const key of Object.keys(modelCardInfo)) {
-    const typedKey = key as keyof ModelCardInfo;
-
-    if (typedKey === 'isPublished') {
+  for (const key of getTypedKeys(modelCardInfo)) {
+    if (key === 'isPublished') {
       continue;
-    } else if (typedKey === 'exampleTopics') {
+    } else if (key === 'exampleTopics') {
       if (
         !modelCardInfo['exampleTopics'].filter(topic => topic.length).length
       ) {
         return false;
       }
-    } else if (!modelCardInfo[typedKey].length) {
+    } else if (!modelCardInfo[key].length) {
       return false;
     }
   }
 
   return true;
 };
 
+const allFieldsHidden = (
+  fieldVisibilities: AichatState['fieldVisibilities']
+) => {
+  for (const key of getTypedKeys(fieldVisibilities)) {
+    if (fieldVisibilities[key] !== Visibility.HIDDEN) {
+      return false;
+    }
+  }
+  return true;
```

Comment:
Not a required change, but just noting that this could be condensed down into a single `every` call:

```
return getTypedKeys(fieldVisibilities).every(key => fieldVisibilities[keys] === Visibility.HIDDEN)
```

If every field is hidden, then it returns `true`, and if any of them are _not_ hidden, it'll return false.

Note that `every` is also smart enough to bail out early if it hits a failure - so if the first element is not hidden, it'll stop immediately and not test the full array.